src/sql/src/plan/transform_ast.rs

```diff
@@ -392,6 +406,53 @@ impl<'ast> VisitMut<'ast, Aug> for FuncRewriter<'_> {
         }
     }
 
+    fn visit_table_with_joins_mut(&mut self, item: &'ast mut TableWithJoins<Aug>) {
+        visit_mut::visit_table_with_joins_mut(self, item);
+        match &mut item.relation {
+            TableFactor::Function {
+                function,
+                alias,
+                with_ordinality,
+            } => {
+                self.rewriting_table_factor = true;
+                // Functions that get rewritten must be rewritten as exprs
+                // because their catalog functions cannot be planned.
+                if let Some((ident, expr)) = self.rewrite_function(function) {
+                    let mut select = Select::default().project(SelectItem::Expr {
+                        expr,
+                        alias: Some(match &alias {
+                            Some(TableAlias { name, columns, .. }) => {
+                                columns.get(0).unwrap_or(name).clone()
+                            }
+                            None => ident,
+                        }),
+                    });
+
+                    if *with_ordinality {
+                        select = select.project(SelectItem::Expr {
+                            expr: Expr::Value(Value::Number("1".into())),
+                            alias: Some("ordinality".into()),
+                        });
+                    }
+
+                    item.relation = TableFactor::Derived {
+                        lateral: false,
+                        subquery: Box::new(Query {
+                            ctes: mz_sql_parser::ast::CteBlock::Simple(vec![]),
+                            body: mz_sql_parser::ast::SetExpr::Select(Box::new(select)),
+                            order_by: vec![],
+                            limit: None,
+                            offset: None,
+                        }),
+                        alias: alias.clone(),
+                    }
```

Comment:
These lines _might_ not be covered at all by any test case (neither unit nor end-to-end test). Would it make sense to add a test for these?

I wrote "might" because the coverage run was conducted on commit `2f5d198` after which (rewriting) changes occurred. Please ignore if changes to the slt test cover this in the mean time.